addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -964,6 +971,9 @@ export class PosStore extends Reactive {
             this.preSyncAllOrders(orders);
             const context = this.getSyncAllOrdersContext(orders);
 
+            if (this.pendingOrder.delete.size) {
```

Comment:
Why not add use the changes made to `getPendingOrder`. You retrieve `orderToDelete` from `this.getPendingOrder()` and do something like `await this.deleteOrders([], orderToDelete.map((o) => o.id));`